### PR TITLE
Totally Mergeable exit-on-fail implementation

### PR DESCRIPTION
**Pull request by itdaniher**
Okay, I mostly fixed this up!

This PR exists to terminate auxin_cli with a non-zero status code if networking failures occur. Without this code, auxin_cli hangs instead of closes nicely.

### CODE REVIEW

File: auxin_cli/Cargo.toml [hunk 1]
```diff
@@ -47,7 +47,7 @@ tracing-subscriber = "0.2.25"
 tracing-futures = { version = "0.2.5", features = ["futures-03", "tokio"] }
 
 tokio-native-tls = "0.3.0"
-tokio = { version = "1.6.1", features = ["full"] }
+tokio = { version = "1.16.1", features = ["full"] }
```
Comment (itdaniher) [inline auxin_cli/Cargo.toml:50]: needed for oneshot blocking_recv

File: auxin_cli/src/main.rs [hunk 2]
```diff
@@ -89,8 +89,33 @@ pub fn launch_repl(_app: &mut crate::app::App) -> Result<()> {
 	panic!("Attempted to launch a REPL, but the 'repl' feature was not enabled at compile-time!")
 }
 
-#[tokio::main(flavor = "current_thread")]
```
Comment (itdaniher) [inline auxin_cli/src/main.rs:92]: this is a simple macro but it's some hiddenness that I couldn't get behind

I read the implementation to figure out what it does

File: auxin_cli/src/main.rs [hunk 3]
```diff
@@ -327,29 +349,34 @@ pub async fn main() -> Result<()> {
 			) = mpsc::channel(MESSAGE_BUF_COUNT);
 
 			tokio::task::spawn_blocking(move || {
```
Comment (itdaniher) [inline auxin_cli/src/main.rs:351]: This just moves the work into the success case so I don't have to deal with possibly uninitialized values. Kinda gross indents but :shrug:

File: auxin_cli/Cargo.toml (hunk 1 above)
Comment (NotGyro) [reply auxin_cli/Cargo.toml:50]: I approve of this version bump, sounds good.

**Review by NotGyro — APPROVED** [review]
All of these changes look reasonable and this will be a useful feature.

**Pull request merged**